concepts/iterators/links.json

```diff
@@ -0,0 +1,18 @@
+[
+    {
+        "url": "https://doc.rust-lang.org/rust-by-example/flow_control/for.html",
+        "description": "examples of using iterators"
+    },
+    {
+        "url": "https://doc.rust-lang.org/book/ch13-02-iterators.html",
+        "description": "official book's chapter on iterators"
+    },
+    {
+        "url": "https://stackoverflow.com/questions/34733811/what-is-the-difference-between-iter-and-into-iter/34745885#34745885",
+        "description": "difference between iter and into_iter"
+    },
+    {
+        "url": "https://doc.rust-lang.org/book/ch13-04-performance.html",
+        "description": "iterator is as or more performant than for loops"
+    }
+]
```

Comment:
Worth adding the official `Iterator` trait documentation: <https://doc.rust-lang.org/std/iter/trait.Iterator.html>.